src/librekey/rnp_key_store.cpp

```diff
@@ -442,25 +449,24 @@ rnp_key_store_add_subkey(rnp_key_store_t *keyring, pgp_key_t *srckey, pgp_key_t
         try {
             keyring->keys.emplace_back();
             oldkey = &keyring->keys.back();
-            keyring->keybyfp[pgp_key_get_fp(srckey)] = std::prev(keyring->keys.end());
+            keyring->keybyfp[srckey->fp()] = std::prev(keyring->keys.end());
```

Comment:
Does this mean, if adding a mapping causes allocation exception, the `keyring->keys` will have an uninitialized element at the end?

Reply:
@rrrooommmaaa Agree, thanks for spotting this. Fixed (and the one below as well).